wgpu-native/src/command/compute.rs

```diff
@@ -29,23 +44,47 @@ pub extern "C" fn wgpu_compute_pass_end_pass(pass_id: ComputePassId) -> CommandB
     pass.cmb_id.value
 }
 
+#[no_mangle]
+pub extern "C" fn wgpu_compute_pass_dispatch(pass_id: ComputePassId, x: u32, y: u32, z: u32) {
+    unsafe {
+        HUB.compute_passes
+            .write()
+            .get_mut(pass_id)
+            .raw
+            .dispatch([x, y, z]);
+    }
+}
+
 #[no_mangle]
 pub extern "C" fn wgpu_compute_pass_set_bind_group(
     pass_id: ComputePassId,
     index: u32,
     bind_group_id: BindGroupId,
 ) {
+    let mut pass_guard = HUB.compute_passes.write();
+    let pass = pass_guard.get_mut(pass_id);
     let bind_group_guard = HUB.bind_groups.read();
-    let set = &bind_group_guard.get(bind_group_id).raw;
-    let layout = unimplemented!();
-    // see https://github.com/gpuweb/gpuweb/pull/93
+    let bind_group = bind_group_guard.get(bind_group_id);
 
-    unsafe {
-        HUB.compute_passes
-            .write()
-            .get_mut(pass_id)
-            .raw
-            .bind_compute_descriptor_sets(layout, index as usize, iter::once(set), &[]);
+    CommandBuffer::insert_barriers(
```

Comment:
To help my understanding could you clarify why the barrier insertion differs between render and compute passes in their `x_set_bind_group` and `x_pass_end`? I thought the granularity would be the same (around the render or compute pass)